### ISSUE TITLE
Bug in aws_s3_bucket resource redirect_all_requests_to action

### ISSUE THREAD

**Issue by fromonesrc** [hashicorp/terraform#5142]
```
resource "aws_s3_bucket" "bucket" {
    bucket = "foo"

    website {
        redirect_all_requests_to = "https://www.foo.com/bar"
    }
}
```

results in this apply

```
aws_s3_bucket.bucket: Creating...
  acl:                                "" => "private"
  arn:                                "" => "<computed>"
  bucket:                             "" => "foo"
  force_destroy:                      "" => "0"
  hosted_zone_id:                     "" => "<computed>"
  region:                             "" => "<computed>"
  website.#:                          "" => "1"
  website.0.redirect_all_requests_to: "" => "https://www.foo.com/bar"
  website_domain:                     "" => "<computed>"
  website_endpoint:                   "" => "<computed>"
```

but the actual value set on S3 drops everything after `.com`

![2016-02-15 at 2 28 42 pm](https://cloud.githubusercontent.com/assets/550256/13058331/8266eec2-d3f0-11e5-997d-22976103a31e.png)

I am running v0.6.11.

**Comment by jbiel** [comment]
+1, same thing observed here. Subsequent runs of terraform indicate that it's trying to update to the proper URL. I've tried escaping, using double slashes, a trailing slash, etc. but nothing seems to work.

```
terraform --version
Terraform v0.6.12-dev (2cfdfd1d84620d4571e6377796ad9f1d96231993)
...
aws_s3_bucket.bucket: Modifying...
  website.0.redirect_all_requests_to: "https://www.foo.com" => "https://www.foo.com/bar"
aws_s3_bucket.bucket: Modifications complete
```

**Comment by doobix** [comment]
+1, same problem here!

**Comment by jbiel** [comment]
I've opened a PR for this: https://github.com/hashicorp/terraform/pull/7883

Long story short is that the [S3 API parameter HostName](http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketPUTwebsite.html#RESTBucketPUTwebsite-requests-request-elements) (which is used for this functionality) is poorly named because it can accept more than just a hostname (it can contain a path too.) The terraform code was treating it as it only accepted hostnames; who can blame them?

**Comment by stack72** [comment]
Closed via #7883
